script.js

```diff
@@ -1,4 +1,318 @@
+  }
+};
+
+var generatePlayerDecks = function () {
+  numberOfPlayer = numPlayers.value;
+  generatePlayers();
+  shuffleDeck();
+  shuffledDeck = cardDeck;
+  for (var i = 0; i < allPlayerCard.length; i += 1) {
+    currentPlayer = i;
+    allPlayerCard[i].card[0] = shuffledDeck.pop();
+    updateNumberOfAce();
+    allPlayerCard[i].card[1] = shuffledDeck.pop();
+    updateNumberOfAce();
+  }
+  currentPlayer = 0;
+  numPlayers.disabled = true;
+};
+
+var updateNumberOfAce = function () {
+  var playerNumberOfCards = allPlayerCard[currentPlayer].card.length;
+  if (
+    allPlayerCard[currentPlayer].card[playerNumberOfCards - 1].name == "ace"
+  ) {
+    allPlayerCard[currentPlayer].numberOfAce += 1;
+  }
+};
+
+var checkIfBlackJack = function () {
+  for (var i = 0; i < allPlayerCard.length; i += 1) {
+    var firstCard = allPlayerCard[i].card[0];
+    var secCard = allPlayerCard[i].card[1];
+    if (firstCard.name == "ace" || secCard.name == "ace") {
+      if (firstCard.value == 10 || secCard.value == 10) {
+        allPlayerCard[i].blackJack = true;
+      }
+    }
+  }
+};
+
+var checkInitialWinning = function () {
+  checkIfBlackJack();
+  var outputMsg = "";
+  var dealerCard = allPlayerCard[allPlayerCard.length - 1];
+  if (dealerCard.blackJack == true) {
+    outputMsg += `Dealer got BlackJack! <br>`;
+    for (var i = 0; i < allPlayerCard.length - 1; i += 1) {
+      if (allPlayerCard[i].blackJack == true) {
+        outputMsg += `${allPlayerCard[i].name} got BlackJack as well! <br>`;
+      } else outputMsg += `${allPlayerCard[i].name} lost! <br>`;
+    }
+    hit.disabled = true;
+    stay.disabled = true;
+    reset.style.visibility = "visible";
+  } else if (
+    allPlayerCard.length == 2 &&
+    allPlayerCard[currentPlayer].blackJack == true
+  ) {
+    outputMsg = `${allPlayerCard[currentPlayer].name} got BlackJack!`;
+    hit.disabled = true;
+    stay.disabled = true;
+    reset.style.visibility = "visible";
+  } else if (allPlayerCard[currentPlayer].blackJack == true) {
+    outputMsg = `${allPlayerCard[currentPlayer].name} got BlackJack!`;
+    currentPlayer += 1;
+  }
+  return outputMsg;
+};
+
+var updateCardTotal = function () {
+  var playerCard = allPlayerCard[currentPlayer].card;
+  var playerNumberOfAce = allPlayerCard[currentPlayer].numberOfAce;
+  var playerNumberOfCards = allPlayerCard[currentPlayer].card.length;
+  allPlayerCard[currentPlayer].cardTotal[0] = 0;
+  allPlayerCard[currentPlayer].cardTotal[1] = 0;
+  for (var i = 0; i < playerCard.length; i += 1) {
+    allPlayerCard[currentPlayer].cardTotal[0] += playerCard[i].value;
+    allPlayerCard[currentPlayer].cardTotal[1] += playerCard[i].value;
+  }
+  if (playerNumberOfAce == 1 && playerNumberOfCards == 2) {
+    allPlayerCard[currentPlayer].cardTotal[1] += 10;
+  }
+  if (playerNumberOfAce == 1 && playerNumberOfCards == 3) {
+    allPlayerCard[currentPlayer].cardTotal[1] += 9;
+  }
+  if (playerNumberOfAce == 2 && playerNumberOfCards == 3) {
```

Comment:
I think these ace conditions are mutually exclusive, probably can use `else if` to reduce unnecessary computing